### DISCUSSION TITLE
How to correctly configure the dap of c++?

### DISCUSSION THREAD

**Discussion by TwelveNight** [LazyVim/LazyVim#924] [Q&A] [answered]
I used cpptools, nvim-dap found it, but when I debugged, I got an error: 
```shell
Error on launch: Unable to start debugging. Launch options string provided by the project system is invalid. Unable to determine path to debugger. Please specify the "MIDebuggerPath" option.
```
I didn't Configure MIDebuggerPath, and I don't know how to configure
I looked at LazyVim's configuration of go, and it seemed that there was no such content at the time

I just downloaded cpptools using Mason, and added the following two pieces of code to lazy.lua

```lua
   { import = "lazyvim.plugins.extras.dap.core" },
   { import = "lazyvim.plugins.extras.dap.nlua" },
```

I just added these codes without any other configuration

![QQ截图20230613165536](https://github.com/LazyVim/LazyVim/assets/113519679/f4d7e5f9-6b7e-4c84-87b3-9bf2c24ae273)

This is the execution path
![QQ截图20230613170755](https://github.com/LazyVim/LazyVim/assets/113519679/a4cbb8a0-466a-4c03-b099-a9941b185db5)

**Comment by jyuan0** [comment, marked as answer]
`cpptools` was automatically configured by [mason-nvim-dap.nvim](https://github.com/jay-babu/mason-nvim-dap.nvim). Maybe you don't have `gdb` installed and on your `PATH`? - https://github.com/jay-babu/mason-nvim-dap.nvim/blob/main/lua/mason-nvim-dap/mappings/configurations.lua#L168

**Reply by TwelveNight** [reply]
thank you for your help
